File: programmers/모의고사.py
answers = [1,2,3,4,5]
def solution(answers):
    answer = []
    one, two, tre,a,b,c= [1,2,3,4,5]*((len(answers)//5)+1), [2,1,2,3,2,4,2,5]*((len(answers)//8)+1), [3,3,1,1,2,2,4,4,5,5]*((len(answers)//10)+1),0,0,0
    for n in range(len(answers)): 
        if one[n] == answers[n]: a += 1
        if two[n] == answers[n]: b += 1
        if tre[n] == answers[n]: c += 1
    x = max(a,b,c)
    if a==x: answer.append(1)
    if b==x: answer.append(2)   
    if c==x: answer.append(3)
    sorted(answer)
    return answer

# def solution(answers):
#     p = [[1, 2, 3, 4, 5],
#          [2, 1, 2, 3, 2, 4, 2, 5],
#          [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
#     s = [0] * len(p)

#     for q, a in enumerate(answers):
#         for i, v in enumerate(p):
#             if a == v[q % len(v)]:
#                 s[i] += 1
#     return [i + 1 for i, v in enumerate(s) if v == max(s)]

File: 프로그래머스/lv1/42840. 모의고사/모의고사.go
func solution(answers []int) (result []int ){
	answer := []int {0,0,0}
	candi := [][]int {
		{1,2,3,4,5},
		{2, 1, 2, 3, 2, 4, 2, 5},
		{3, 3, 1, 1, 2, 2, 4, 4, 5, 5},
	}
	for i:=0; i<len(answers); i++ {
		for j:=0; j<3; j++ {
			if answers[i] == candi[j][i%len(candi[j])] {
				answer[j] += 1
			}
		}
	}
	max := -1
	for i:=0; i<3; i++ {
		if answer[i] > max {
			result = []int {i+1}
			max = answer[i]
		} else if answer[i] == max {
			result = append(result, i+1)
		}
	}
	return
}